# PROJET SEATTLE ENERGY BENCHMARKING
## Notebook 00 : initialisation de l'environnement et ingestion des données

---

### Identité du document
* **Statut :** Phase 1 (exploration & prototypage)
* **Date de création :** 26 Décembre 2025
* **Dernière mise à jour :** 29 Décembre 2025
* **Dépendances notebooks**: Aucune

### Description
Ce notebook établit les fondations du projet. Il assure la mise en place de l'infrastructure et la configuration des outils de gestion . L'objectif est de garantir une base reproductible avant d'entamer l'audit de qualité.

### Objectifs principaux
1. Initialiser le système de gestion de configuration via Hydra.
2. Déployer l'arborescence des répertoires du projet.
3. Récupérer et charger le jeu de données source (Immuable).
4. Établir un premier diagnostic structurel des données chargées.

### Dépendances critiques
* `hydra` : Gestion de la configuration.
* `src.utils` & `src.data` : Modules internes de support.

### LIVRABLES
1. Environnement technique initialisé (Hydra, Dossiers).
2. Dataset brut dans `data/raw/`.
3. Dictionnaire de données initial (`reports/data_dictionary.md`).
4. Note de synthèse structurelle.
5. Notebook propre

---

# Table des Matières
- [Section 0 : Importation des packages ](#section-1--configuration)
- [Section 1 : Configuration et initialisation ](#section-1--configuration)
- [Section 2 : Contextualisation & provenance](#section-2--contextualisation--provenance)
- [Section 3 : Chargement et inspection structurelle](#section-3--ingestion-et-premier-contact)
- [Section 4 : Typologie des variables](#section-4--typologie-et-cartographie-des-variables)
- [Section 5 : Focus sur la variable cible](#section-5--analyse-de-la-variable-cible)
- [Section 6 : Identification des redondances apparentes](#section-6--étude-des-redondances-et-unités)
- [Section 7 :Premières incohérences détectées](#section-7--détection-des-signaux-dalerte-immédiats)
- [Section 8 : Synthèse et génération du dictionnaire](#section-8--synthèse-et-génération-du-dictionnaire)
---

# Section 0 : Importation des packages

In [4]:
import logging
import pandas as pd
from pathlib import Path


# Import des fonctions utilitaires 

import sys
from pathlib import Path

PROJECT_ROOT = Path.cwd().parent
SRC_PATH = PROJECT_ROOT / "src"

if str(SRC_PATH) not in sys.path:
    sys.path.insert(0, str(SRC_PATH))


from data.load_data import load_data_raw
from utils.config_loader import load_config, create_directories
from utils.eda_logger import setup_eda_logger
 


# Section 1:Configuration et initialisation

In [5]:
# Chargement de la configuration principale (Hydra)
cfg = load_config()

# Initialisation du logger
setup_eda_logger(cfg)
logger = logging.getLogger(__name__)

# Création des dossiers nécessaires à l'exécution (raw, interim, processed, reports, etc.)
create_directories(cfg)


2025-12-30 20:34:03,872 - utils.config_loader - INFO - Configuration 'config' chargée (project_root=C:\Users\HP\Desktop\temp\TODO\SEMESTRE_1\ML1\ML-prediction-CO2)
2025-12-30 20:34:03,875 - utils.config_loader - INFO - Répertoire prêt : C:\Users\HP\Desktop\temp\TODO\SEMESTRE_1\ML1\ML-prediction-CO2\data\raw
2025-12-30 20:34:03,876 - utils.config_loader - INFO - Répertoire prêt : C:\Users\HP\Desktop\temp\TODO\SEMESTRE_1\ML1\ML-prediction-CO2\data\interim
2025-12-30 20:34:03,877 - utils.config_loader - INFO - Répertoire prêt : C:\Users\HP\Desktop\temp\TODO\SEMESTRE_1\ML1\ML-prediction-CO2\data\processed
2025-12-30 20:34:03,877 - utils.config_loader - INFO - Répertoire prêt : C:\Users\HP\Desktop\temp\TODO\SEMESTRE_1\ML1\ML-prediction-CO2\figures
2025-12-30 20:34:03,877 - utils.config_loader - INFO - Répertoire prêt : C:\Users\HP\Desktop\temp\TODO\SEMESTRE_1\ML1\ML-prediction-CO2\reports


# Section 2 : Chargement et inspection structurelle

Cette section se concentre sur le chargement initial des données à partir du fichier CSV et une inspection structurelle de base. L'objectif est d'obtenir une première vue d'ensemble du dataset, de vérifier son intégrité et d'identifier tout problème évident au niveau du format ou de la structure.

## Importation du CSV

In [6]:
# Chargement des données brutes
df_raw = load_data_raw(cfg)

# Validation dimensions
n_rows, n_cols = df_raw.shape
logger.info(f"Dataset chargé ({n_rows} lignes, {n_cols} colonnes)")



2025-12-30 20:43:27,072 - data.load_data - INFO - DataFrame chargé : 3376 lignes, 46 colonnes
2025-12-30 20:43:27,102 - data.load_data - INFO -  Intégrité des données validée (Aucune modification détectée).
2025-12-30 20:43:27,106 - __main__ - INFO - Dataset chargé (3376 lignes, 46 colonnes)


---
Le chargement nous donne un DataFrame avec 3376 lignes et 46 colonnes.

## Affichage pour première impression visuelle

In [ ]:
# Forcer l'affichage de toutes les colonnes 
pd.set_option("display.max_columns", None)
# Affichage des premières lignes
print("Premières 10 lignes :")
display(df_raw.head(10))

# Affichage des dernières lignes
print("Dernières 10 lignes :")
display(df_raw.tail(10))

# Affichage d'un échantillon aléatoire
print("Échantillon aléatoire de 20 lignes :")
display(df_raw.sample(20))

OSEBuildingID  DataYear          BuildingType   PrimaryPropertyType  \
0              1      2016        NonResidential                 Hotel   
1              2      2016        NonResidential                 Hotel   
2              3      2016        NonResidential                 Hotel   
3              5      2016        NonResidential                 Hotel   
4              8      2016        NonResidential                 Hotel   
5              9      2016    Nonresidential COS                 Other   
6             10      2016        NonResidential                 Hotel   
7             11      2016        NonResidential                 Other   
8             12      2016        NonResidential                 Hotel   
9             13      2016  Multifamily MR (5-9)  Mid-Rise Multifamily   

                  PropertyName          Address     City State  ZipCode  \
0         Mayflower park hotel    405 Olive way  Seattle    WA  98101.0   
1              Paramount Hotel  724 Pine street  Seattle    WA  98101.0   
2      5673-The Westin Seattle  1900 5th Avenue  Seattle    WA  98101.0   
3                    HOTEL MAX   620 STEWART ST  Seattle    WA  98101.0   
4  WARWICK SEATTLE HOTEL (ID8)    401 LENORA ST  Seattle    WA  98121.0   
5                West Precinct  810 Virginia St  Seattle    WA  98101.0   
6                       Camlin  1619 9th Avenue  Seattle    WA  98101.0   
7            Paramount Theatre      911 Pine St  Seattle    WA  98101.0   
8         311wh-Pioneer Square      612 2nd Ave  Seattle    WA  98104.0   
9                Lyon Building   607 - 3rd Ave.  Seattle    WA  98104.0   

  TaxParcelIdentificationNumber  CouncilDistrictCode Neighborhood  Latitude  \
0                    0659000030                    7     DOWNTOWN  47.61220   
1                    0659000220                    7     DOWNTOWN  47.61317   
2                    0659000475                    7     DOWNTOWN  47.61393   
3                    0659000640                    7     DOWNTOWN  47.61412   
4                    0659000970                    7     DOWNTOWN  47.61375   
5                    0660000560                    7     DOWNTOWN  47.61623   
6                    0660000825                    7     DOWNTOWN  47.61390   
7                    0660000955                    7     DOWNTOWN  47.61327   
8                    0939000080                    7     DOWNTOWN  47.60294   
9                    0939000105                    7     DOWNTOWN  47.60284   

   Longitude  YearBuilt  NumberofBuildings  NumberofFloors  PropertyGFATotal  \
0 -122.33799       1927                1.0              12             88434   
1 -122.33393       1996                1.0              11            103566   
2 -122.33810       1969                1.0              41            956110   
3 -122.33664       1926                1.0              10             61320   
4 -122.34047       1980                1.0              18            175580   
5 -122.33657       1999                1.0               2             97288   
6 -122.33283       1926                1.0              11             83008   
7 -122.33136       1926                1.0               8            102761   
8 -122.33263       1904                1.0              15            163984   
9 -122.33184       1910                1.0               6             63712   

   PropertyGFAParking  PropertyGFABuilding(s)  \
0                   0                   88434   
1               15064                   88502   
2              196718                  759392   
3                   0                   61320   
4               62000                  113580   
5               37198                   60090   
6                   0                   83008   
7                   0                  102761   
8                   0                  163984   
9                1496                   62216   

               ListOfAllPropertyUseTypes  \
0                          

Dernières 10 lignes :


OSEBuildingID  DataYear        BuildingType PrimaryPropertyType  \
3366          50210      2016  Nonresidential COS              Office   
3367          50212      2016  Nonresidential COS               Other   
3368          50219      2016  Nonresidential COS  Mixed Use Property   
3369          50220      2016  Nonresidential COS              Office   
3370          50221      2016  Nonresidential COS               Other   
3371          50222      2016  Nonresidential COS              Office   
3372          50223      2016  Nonresidential COS               Other   
3373          50224      2016  Nonresidential COS               Other   
3374          50225      2016  Nonresidential COS  Mixed Use Property   
3375          50226      2016  Nonresidential COS  Mixed Use Property   

                             PropertyName              Address     City State  \
3366         Central West HQ / Brown Bear          1403 w howe  Seattle    WA   
3367                  Conservatory Campus      1400 E Galer St  Seattle    WA   
3368            Garfield Community Center  2323 East Cherry St  Seattle    WA   
3369                     Genesee/SC SE HQ       4420 S Genesee  Seattle    WA   
3370          High Point Community Center     6920 34th Ave SW  Seattle    WA   
3371                Horticulture building     1600 S Dakota St  Seattle    WA   
3372  International district/Chinatown CC        719 8th Ave S  Seattle    WA   
3373                      Queen Anne Pool       1920 1st Ave W  Seattle    WA   
3374          South Park Community Center       8319 8th Ave S  Seattle    WA   
3375          Van Asselt Community Center     2820 S Myrtle St  Seattle    WA   

      ZipCode TaxParcelIdentificationNumber  CouncilDistrictCode  \
3366      NaN                    2425039137                    7   
3367      NaN                    2925049087                    3   
3368      NaN                    7544800245                    3   
3369      NaN                    4154300585                    2   
3370      NaN                    2524039059                    1   
3371      NaN                    1624049080                    2   
3372      NaN                    3558300000                    2   
3373      NaN                    1794501150                    7   
3374      NaN                    7883603155                    1   
3375      NaN                    7857002030                    2   

                Neighborhood  Latitude  Longitude  YearBuilt  \
3366   MAGNOLIA / QUEEN ANNE  47.63572 -122.37525       1952   
3367                    EAST  47.63228 -122.31574       1912   
3368                 CENTRAL  47.60775 -122.30225       1994   
3369               SOUTHEAST  47.56440 -122.27813       1960   
3370  DELRIDGE NEIGHBORHOODS  47.54067 -122.37441       1982   
3371        GREATER DUWAMISH  47.56722 -122.31154       1990   
3372                DOWNTOWN  47.59625 -122.32283       2004   
3373   MAGNOLIA / QUEEN ANNE  47.63644 -122.35784       1974   
3374        GREATER DUWAMISH  47.52832 -122.32431       1989   
3375        GREATER DUWAMISH  47.53939 -122.29536       1938   

      NumberofBuildings  NumberofFloors  PropertyGFATotal  PropertyGFAParking  \
3366                1.0               1             13661                   0   
3367                1.0               1             23445                   0   
3368                1.0               1             20050                   0   
3369                1.0               1             15398                   0   
3370                1.0               1             18261                   0   
3371                1.0               1             12294                   0   
3372                1.0               1             16000                   0   
3373                1.0               1             13157                   0   
3374                1.0               1             14101                   0   
3375                1.0               1       

Échantillon aléatoire de 20 lignes :


OSEBuildingID  DataYear          BuildingType    PrimaryPropertyType  \
2227          24868      2016        NonResidential       Worship Facility   
1119          20928      2016        NonResidential       Worship Facility   
2780          26919      2016  Multifamily LR (1-4)   Low-Rise Multifamily   
1516          22194      2016  Multifamily LR (1-4)   Low-Rise Multifamily   
6                10      2016        NonResidential                  Hotel   
1913          23789      2016  Multifamily HR (10+)  High-Rise Multifamily   
710           19682      2016  Multifamily MR (5-9)   Mid-Rise Multifamily   
3367          50212      2016    Nonresidential COS                  Other   
1015          20512      2016  Multifamily LR (1-4)   Low-Rise Multifamily   
2080          24381      2016    Nonresidential COS     Mixed Use Property   
1724          23223      2016  Multifamily MR (5-9)   Mid-Rise Multifamily   
1370          21649      2016  Multifamily LR (1-4)   Low-Rise Multifamily   
1824          23518      2016  Multifamily LR (1-4)   Low-Rise Multifamily   
2145          24593      2016        NonResidential  Self-Storage Facility   
1711          23182      2016  Multifamily MR (5-9)   Mid-Rise Multifamily   
660           19489      2016        NonResidential              Warehouse   
502             683      2016        NonResidential                  Other   
3016          27885      2016  Multifamily MR (5-9)   Mid-Rise Multifamily   
308             439      2016        NonResidential     Mixed Use Property   
289             419      2016                Campus     Mixed Use Property   

                      PropertyName                 Address     City State  \
2227                      Magnolia           2415 31st Ave  Seattle    WA   
1119      St. Paul Church & School       10001 57th Ave. S  Seattle    WA   
2780              Lambda Chi Alpha        4509 19th Ave NE  Seattle    WA   
1516                    Sixty Five            1433 NW 64th  Seattle    WA   
6                           Camlin         1619 9th Avenue  Seattle    WA   
1913       Highlander Condominiums   525 Belmount Ave East  Seattle    WA   
710         The Audrey at Belltown     2922 Western Avenue  Seattle    WA   
3367           Conservatory Campus         1400 E Galer St  Seattle    WA   
1015             Central Park East       2001 E Yesler Way  Seattle    WA   
2080      Rainier Community Center         4630 38th Ave S  Seattle    WA   
1724                 Hudson Square  4852 California Ave SW  Seattle    WA   
1370     Belroy Apartments Vintage      703 Bellevue Ave E  Seattle    WA   
1824           Serana Condominiums      621 5th Ave. North  Seattle    WA   
2145          1st Ave Self Storage          2400 1st Ave S  Seattle    WA   
1711             Landes Apartments          901 8th Avenue  Seattle    WA   
660            Seadrunar Recycling         28 S Brandon St  Seattle    WA   
502           Rainier Square Plaza      3820 Rainier Ave S  Seattle    WA   
3016                     Lee Plaza        4746 11th Ave NE  Seattle    WA   
308   Fisher Plaza - East Building           140 4th Ave N  Seattle    WA   
289                    Main Market         1501 Pike Place  Seattle    WA   

      ZipCode TaxParcelIdentificationNumber  CouncilDistrictCode  \
2227  98199.0                    8127700530                    7   
1119  98178.0                    1686400005                    2   
2780  98105.0                    8823902740                    4   
1516  98107.0                    2767600820                    6   
6     98101.0                    0660000825                    7   
1913  98102.0                    3302700000                    3   
710   98121.0                    0695000005                    7   
3367      NaN                    2925049087                    3   
1015  98122.0                    1496130000                    3   
2080  98118.0                    7950304230                    2   
1